sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -582,6 +582,15 @@ object SQLConf {
     .intConf
     .createWithDefault(2667)
 
+  val CODEGEN_MAX_CHARS_PER_FUNCTION = buildConf("spark.sql.codegen.maxCharactersPerFunction")
```

Comment:
To be honest, using the number of characters looks pretty fragile. It is even worse than using the number of lines.